src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -197,4 +209,21 @@ export class ClusterClient {
       throw new Error('Cluster client cannot be used after it has been closed.');
     }
   }
+
+  // supports both KibanaRequest & Legacy.Request while asScoped is used in LegacyPlatform
+  private getHeaders(
+    request?: KibanaRequest | Request | FakeRequest
+  ): Record<string, string | string[] | undefined> {
+    if (isValidRequest(request)) {
```

Comment:
I don't like we have to this, but reporting and upgrade_assistant abuse `callWithRequest` via creating `fakeRequest`
https://github.com/restrry/kibana/blob/e0156634c05e4ddcb9e3f4b1c4367ada0f457b5a/x-pack/plugins/upgrade_assistant/server/lib/reindexing/worker.ts#L162
https://github.com/restrry/kibana/blob/e0156634c05e4ddcb9e3f4b1c4367ada0f457b5a/x-pack/plugins/reporting/export_types/csv/server/execute_job.js#L52

we can add deprecation warning that `fakeRequest` support will be removed in New Platform.

plus this helper could be used in other places when we have to [workaround fakeRequests](https://github.com/elastic/kibana/pull/35429#discussion_r288153327)